algorithms/2.TCGA-MLexample-LungCancer.ipynb

```diff
@@ -0,0 +1,311 @@
+    "Y = pd.read_table(path, index_col=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# GENES: AKT1, ALK, BRAF, DDR2, EGFR, ERBB2, KRAS, MAP2K1, NRAS, PIK3CA, PTEN, RET, RIT1, ROS1\n",
+    "\n",
+    "list_of_tsgenes = [207,238,673,4921,1956,2064,3845,5604,4893,5290,5728,5979,6016,6098]\n",
+    "\n",
+    "AUROCS_train = []\n",
+    "AUROCS_test = []\n",
+    "p = []"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n",
+      "Wall time: 0 ns\n"
+     ]
+    }
+   ],
+   "source": [
+    "for GENE in list_of_tsgenes:\n",
+    "    \n",
+    "    y = Y[str(GENE)]\n",
+    "    # Here are the percentage of tumors with NF1\n",
+    "    per = y.value_counts(True)\n",
+    "    # Set aside 10% of the data for testing, this can only be done where the number of mutations is large enough\n",
+    "    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)\n",
+    "    'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))\n",
+    "    def fs_mad(x, y):\n",
+    "        \"\"\"    \n",
+    "        Get the median absolute deviation (MAD) for each column of x\n",
+    "        \"\"\"\n",
+    "        scores = mad(x) \n",
+    "        return scores, np.array([np.NaN]*len(scores))\n",
+    "\n",
+    "    # select the top features with the highest MAD\n",
+    "    feature_select = SelectKBest(fs_mad, k=n_feature_kept)\n",
+    "    # Include loss='log' in param_grid doesn't work with pipeline somehow\n",
+    "    clf = SGDClassifier(random_state=0, class_weight='balanced',\n",
+    "                    loss=param_fixed['loss'], penalty=param_fixed['penalty'])\n",
+    "\n",
+    "    # joblib is used to cross-validate in parallel by setting `n_jobs=-1` in GridSearchCV\n",
+    "    # Supress joblib warning. See https://github.com/scikit-learn/scikit-learn/issues/6370\n",
+    "    warnings.filterwarnings('ignore', message='Changing the shape of non-C contiguous array')\n",
+    "    clf_grid = grid_search.GridSearchCV(estimator=clf, param_grid=param_grid, n_jobs=-1, scoring='roc_auc')\n",
+    "    pipeline = make_pipeline(\n",
+    "        feature_select,  # Feature selection\n",
+    "        StandardScaler(),  # Feature scaling\n",
+    "        clf_grid)\n",
+    "    %%time\n",
```

Comment:
I think this `%%time` should be removed -- it doesn't look like it's really working :smile_cat: